File: Quiz_3/Aadrika_Behl_590012053/Quiz_3_Debugging_Question_2.py
def sort_colors(nums):
    left, current, right = 0, 0, len(nums) - 1

    while current <= right:
        if nums[current] == 0:  
            nums[left], nums[current] = nums[current], nums[left]
            left += 1
            current += 1
        elif nums[current] == 1:  
            current += 1
        else: 
            nums[current], nums[right] = nums[right], nums[current]
            right -= 1
           

    return nums

File: Quiz_3/Agam_590011213/Quiz_3_Debugging_Question_2.c
#include <stdio.h>

void sort_colors(int nums[], int size) {
    int left = 0, current = 0, right = size - 1;
    while (current <= right) {
        if (nums[current] == 0) {
            int temp = nums[left];
            nums[left] = nums[current];
            nums[current] = temp;
            left++;
            current++;
        } else if (nums[current] == 1) {
            current++;
        } else {
            int temp = nums[current];
            nums[current] = nums[right];
            nums[right] = temp;
            right--;
        }
    }
}

int main() {
    int nums[] = {2, 0, 2, 1, 1, 0};
    int size = sizeof(nums) / sizeof(nums[0]);
    sort_colors(nums, size);
    printf("Sorted array: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
    return 0;
}
